### ISSUE TITLE
latest nightly toolchain (a37c32e2d 2020-06-11) fails to build this crate: "file with name 'README.md' was not found"

### ISSUE THREAD

**Issue by oconnor663** [XAMPPRocky/remove_dir_all#19]
https://github.com/oconnor663/blake2_simd/runs/764237325

In that run, I see this failure in one of my sub-crates:

```
/usr/share/rust/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo test

    Updating crates.io index
 Downloading crates ...
error: failed to download `remove_dir_all v0.5.2`

Caused by:
  unable to get packages from source

Caused by:
  failed to parse manifest at `/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/remove_dir_all-0.5.2/Cargo.toml`

Caused by:
  readme file with name 'README.md' was not found
Command failed :(
```

In my CI setup, it only fails on the nightly compiler, but it seems to reliably fail with nightly on every OS target. However locally, I can't seem to reproduce the failure at all.

Any chance you recognize this error message? I just have no idea what could really be causing it.

**Comment by XAMPPRocky** [comment]
Thank you for your issue! If you’re only getting this on nightly then likely there’s been regression on the rustc/cargo side. Would you mind reporting it upstream?

**Comment by dignifiedquire** [comment]
It seems this was fixed on `master` in e2747c6b76d49e17175c7b716ef6cbea61e13f2a but not yet published. As far as I understand the issue is that cargo/rustc is now more strict on these checks, actually failing the build when these files are missing.

**Comment by oconnor663** [comment]
Ok, now I see why I wasn't repro'ing this locally. I was using the default rustup profile, so `rustup update` was doing this:

```
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'                                                                                                                           
info: latest update on 2020-06-12, rust version 1.46.0-nightly (a37c32e2d 2020-06-11)                                                                                                          
info: skipping nightly which is missing installed component 'rustfmt-preview'                                                                                                                  
```

But on CI, I was doing `rustup set profile minimal`, so I was able to pick up that build. Presumably in a few days when a more complete nightly build comes out, everyone running the default profile will start seeing this issue on nightly too.

**Comment by oconnor663** [comment]
I just confirmed that adding this to my `Cargo.toml` fixes the problem:

```
[patch.crates-io]
remove_dir_all = { git = 'https://github.com/XAMPPRocky/remove_dir_all' }
```

Based on that, could you publish current master as a `v0.5.3` release? I think that would fix most callers automatically.

**Comment by oconnor663** [comment]
I just filed https://github.com/rust-lang/rust/issues/73282 upstream.

**Comment by XAMPPRocky** [comment]
I’ll publish a patch release shortly that should address the issue.

**Comment by kvark** [comment]
We are also hit by this in gfx-rs land. Looking forward to a patch!

**Comment by XAMPPRocky** [comment]
This has been fixed in `0.5.3`.
